apple/testing/default_runner/ios_xctestrun_runner.template.sh

```diff
@@ -149,14 +174,55 @@ if [[ -n "$test_host_path" ]]; then
     cp -R "$libraries_path/PrivateFrameworks/XCUIAutomation.framework" "$runner_app_frameworks_destination/XCUIAutomation.framework"
     cp -R "$libraries_path/PrivateFrameworks/XCTAutomationSupport.framework" "$runner_app_frameworks_destination/XCTAutomationSupport.framework"
     cp -R "$libraries_path/PrivateFrameworks/XCUnit.framework" "$runner_app_frameworks_destination/XCUnit.framework"
+    cp "$developer_path/usr/lib/libXCTestSwiftSupport.dylib" "$runner_app_frameworks_destination/libXCTestSwiftSupport.dylib"
+    cp "$developer_path/usr/lib/libXCTestBundleInject.dylib" "$runner_app_frameworks_destination/libXCTestBundleInject.dylib"
     # Added in Xcode 14.3
     xctestsupport_framework_path="$libraries_path/PrivateFrameworks/XCTestSupport.framework"
     if [[ -d "$xctestsupport_framework_path" ]]; then
       cp -R "$xctestsupport_framework_path" "$runner_app_frameworks_destination/XCTestSupport.framework"
     fi
+    if [[ "$build_for_device" == true ]]; then
+      # XCTRunner is multi-archs. When launching XCTRunner on arm64e device, it
+      # will be launched as arm64e process by default. If the test bundle is arm64
+      # bundle, the XCTRunner which hosts the test bundle will fail to be
+      # launched. So removing the arm64e arch from XCTRunner can resolve this
+      # case.
+      /usr/bin/lipo "$test_tmp_dir/$runner_app/XCTRunner" -remove arm64e -output "$test_tmp_dir/$runner_app/XCTRunner"
+    fi
+    test_host_mobileprovision_path="$test_tmp_dir_test_host_path/embedded.mobileprovision"
+    # Only engage signing workflow if the test host is signed
+    if [[ -f "$test_host_mobileprovision_path" ]]; then
+      test_runner_mobileprovision_path="$test_tmp_dir/$runner_app/embedded.mobileprovision"
+      cp "$test_host_mobileprovision_path" "$test_runner_mobileprovision_path"
```

Comment:
[nit] line reduction, simplification
```suggestion
      cp "$test_host_mobileprovision_path" "$test_tmp_dir/$runner_app/embedded.mobileprovision"
```